s3/libirods_s3.cpp

```diff
@@ -494,121 +506,126 @@ irods::error s3ReadAuthInfo(
-                msg << "\"";
-            }
+        std::string&& hostname = s3GetHostname(_prop_map);
+        const char* host_name = hostname.c_str();  // Iterate through on each try
+        status = S3_initialize( "s3", flags, host_name );
 
-            result = ASSERT_ERROR(status == S3StatusOK, status, "Error initializing the S3 library. Status = %d.",
-                                  status, msg.str().c_str());
-            if( result.ok() ) {
+        std::stringstream msg;
+        if( status >= 0 ) {
+            msg << " - \"";
+            msg << S3_get_status_name((S3Status)status);
+            msg << "\"";
+        }
 
-                // If using V4 we also need to set the S3 region name
-                if (signature_version == S3SignatureV4) {
-                    std::string region_name = "us-east-1";
+        result = ASSERT_ERROR(status == S3StatusOK, status, "Error initializing the S3 library. Status = %d.",
+                              status, msg.str().c_str());
+        if( result.ok() ) {
 
-                    // Get S3 region name from plugin property map
-                    if (!_prop_map.get< std::string >(s3_region_name, region_name ).ok()) {
-                        rodsLog( LOG_ERROR, "Failed to retrieve S3 region name from resource plugin properties, using 'us-east-1'");
-                    }
+            // If using V4 we also need to set the S3 region name
+            if (signature_version == S3SignatureV4) {
+                std::string region_name = "us-east-1";
 
-                    S3Status status = S3_set_region_name(region_name.c_str());
-                    if (status != S3StatusOK) {
-                        rodsLog(LOG_ERROR, "failed to set region name to %s: %s", region_name.c_str(), S3_get_status_name(status));
-                        return ERROR(S3_INIT_ERROR, "S3_set_region_name() failed.");
-                    }
+                // Get S3 region name from plugin property map
+                if (!_prop_map.get< std::string >(s3_region_name, region_name ).ok()) {
+                    rodsLog( LOG_ERROR, "Failed to retrieve S3 region name from resource plugin properties, using 'us-east-1'");
                 }
 
-                break;
+                S3Status status = S3_set_region_name(region_name.c_str());
+                if (status != S3StatusOK) {
+                    rodsLog(LOG_ERROR, "failed to set region name to %s: %s", region_name.c_str(), S3_get_status_name(status));
+                    return ERROR(S3_INIT_ERROR, "S3_set_region_name() failed.");
+                }
             }
 
-            ctr++;
-
-            s3_sleep( g_retry_wait, 0 );
+            break;
+        }
 
-            rodsLog(
-                LOG_NOTICE,
-                "s3Init - Error in connection, retry count %d",
-                ctr );
+        ctr++;
 
-        } // while
+        s3_sleep( wait_time, 0 );
 
-        if( result.ok() ) {
-            S3Initialized = true;
```

Comment:
Can this global be removed now? It is initialized to false and only checked in one other place which creates a dead codepath.

Reply:
Added back in due to its need with S3_deinitialize().